client/client.go

```diff
@@ -29,32 +28,40 @@ func main() {
 		endpoint := *host + "/echo"
 		req, err := http.NewRequest("POST", endpoint, reader)
 		if err != nil {
-			fmt.Println("request could not be initiated: ", err)
-			return 
+			fmt.Println("could not resolve host: ", err)
+			os.Exit(1)
+			return
```

Comment:
Return is redundant if the code calls `os.Exit` first.  The difference between the two is return will call deferred functions, exit won't.  As such, it is common to put logic that may need defers in a separate function that returns an exit code, and then have main `os.Exit(echoserver())` or something similar.

https://gobyexample.com/exit

https://github.com/kubernetes/kubernetes/blob/062fe2ebf14857524d710f455d7d9c5379f99477/cmd/kubectl/kubectl.go#L49-L51
https://github.com/gravitational/gravity/blob/b03882d1b60f8a26ddf659f7be7332f936c4f3a6/tool/tele/main.go#L35-L39